prog/vm/github_runner.rb

```diff
@@ -184,6 +184,11 @@ def before_run
       # ./env.sh sets some variables for runner to run properly
       ./actions-runner/env.sh
 
+      # The runner script loads environment variables from .env file but ignores
+      # the standard /etc/environment file of the operating system.
+      # https://github.com/actions/runner/issues/1703
+      cat /etc/environment >> ./actions-runner/.env
```

Comment:
Is there any relationship between this and ./actions-runner/env.sh we call above? It seems there are multiple sources of environment variables. Could there be anything else we miss?

Reply:
Also, I think appending /etc/environment might be an implied decision about overriding priority of OS vs user-set environment variables...and in this case, the OS would dominate...is that what we want?

Reply:
> Is there any relationship between this and ./actions-runner/env.sh we call above? It seems there are multiple sources of environment variables. Could there be anything else we miss?

"env.sh" writes some variables to the ".env" file. https://github.com/actions/runner/blob/9e3e57ff90c089641a3a5833c2211841da1a37f8/src/Misc/layoutroot/env.sh

Reply:
> Also, I think appending /etc/environment might be an implied decision about overriding priority of OS vs user-set environment variables...and in this case, the OS would dominate...is that what we want?

I agree with that. User-set environment variables (.env) take precedence over OS environment variables. Appending OS variables to the ".env" file increases their priority unnecessarily. 

I refactored the appending part. Now, I pass the environment variables in a way similar to a k8s controller.